mlx_steering_dump_parser.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) 2020 Mellanox Technologies, Inc.  All rights reserved.
+#
+# This software is available to you under a choice of one of two
+# licenses.  You may choose to be licensed under the terms of the GNU
+# General Public License (GPL) Version 2, available from the file
+# COPYING in the main directory of this source tree, or the
+# OpenIB.org BSD license below:
+#
+#     Redistribution and use in source and binary forms, with or
+#     without modification, are permitted provided that the following
+#     conditions are met:
+#
+#      - Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      - Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials
+#        provided with the distribution.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+# BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Parse tool for SW steering debug dump file
+import argparse
+import csv
+
+import dr_trigger
+from dr_action import *
+from dr_domain import *
+from dr_matcher import *
+from dr_rule import *
+from dr_table import *
+from dr_utilities import *
+
+
+#mapping csv records types to it's relevant parser function
+def dr_csv_rec_type_parser(rec_type):
+    switch = {
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN.value[0]: dr_dump_domain,
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN_INFO_DEV_ATTR.value[0]: dr_dump_domain_info_dev_attr,
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN_INFO_CAPS.value[0]: dr_dump_domain_info_caps,
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN_INFO_VPORT.value[0]: dr_dump_domain_info_vport,
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN_SEND_RING.value[0]: dr_dump_domain_send_ring,
+        dr_dump_rec_type.DR_DUMP_REC_TYPE_DOMAIN_INFO_FLEX_PARSER.value[0]: dr_dump_domain_info_flex_parser,
+
```

Comment:
In my editor this whole file has tabs instead of spaces.
In github GUI I don't see it, but please verify.